### ISSUE TITLE
failed process when running build_sysimage(force=true) despite elevated cmd prompt

### ISSUE THREAD

**Issue by musm** [JuliaLang/julia#18377]
```julia
julia> include("C:\\Julia\\julia-0.5-latest\\share\\julia\\build_sysimg.jl")

julia> build_sysimg(force=true)

ERROR: failed process: Process(`'C:\Users\Mus\.julia\v0.5\WinRPM\deps\usr\x86_64-w64-mingw32\sys-root\mingw\bin\gcc.exe' '-LC:\Julia\Julia-0.5-latest\bin' -shared -ljulia -lssp -o 'C:\Julia\Julia-0.5-latest\bin\..\lib\julia\sys.dll' 'C:\Julia\Julia-0.5-latest\bin\..\lib\julia\sys.o'`, ProcessExited(1)) [1]
 in pipeline_error(::Base.Process) at .\process.jl:616
 in run at .\process.jl:592 [inlined]
 in link_sysimg(::String, ::String, ::Bool) at C:\Julia\julia-0.5-latest\share\julia\build_sysimg.jl:165
 in (::##2#3{Bool,String})() at C:\Julia\julia-0.5-latest\share\julia\build_sysimg.jl:90
 in cd(::##2#3{Bool,String}, ::String) at .\file.jl:48
 in #build_sysimg#1(::Bool, ::Bool, ::Function, ::Void, ::String, ::Void) at C:\Julia\julia-0.5-latest\share\julia\build_sysimg.jl:48
 in (::#kw##build_sysimg)(::Array{Any,1}, ::#build_sysimg, ::Void, ::String, ::Void) at .\<missing>:0 (repeats 2 times)

```

**Comment by tkelman** [comment]
What happens if you run `C:\Users\Mus\.julia\v0.5\WinRPM\deps\usr\x86_64-w64-mingw32\sys-root\mingw\bin\gcc.exe -LC:\Julia\Julia-0.5-latest\bin -shared -ljulia -lssp -o C:\Julia\Julia-0.5-latest\bin\..\lib\julia\sys.dll C:\Julia\Julia-0.5-latest\bin\..\lib\julia\sys.o` from cmd? It might be that you're not allowed to overwrite a loaded dll. A workaround I've used elsewhere is you are allowed to rename the loaded dll and create/move something else in its place, but you'd need to restart for the change to take effect. Maybe `build_sysimg` should be adjusted to do this on windows if the file it's writing already exists (and is the currently loaded sysimg, if there's an easy way to get that).

**Comment by musm** [comment]
That works! What about a very aggressive version that doesn't check if it's the currently loaded sysimg?

If you pass force=true on windows it will always rename the dll;
then replace with linked sysimg.dll; 
delete the old dll;

I'm not sure if the last step will work.

**Comment by tkelman** [comment]
Can't delete the old dll if it's still loaded in memory, from what I recall. So the only tricky part is what name to give the moved one (append the current date and time maybe?), and probably should notify the user where the old one was backed up to.

**Comment by tkelman** [comment]
fixed by #18393 (the comment needs to be in the original PR description for github to auto-close, it doesn't check later comments)
